File: backend/api/schemas/payments.py
"""Request/response schemas for the payments router."""
from pydantic import BaseModel


class LogPaymentRequest(BaseModel):
    """payment_date/week are no longer client-supplied — log_payment()
    determines both itself from the server's real current date."""

    vendor_id: int
    amount: float


class LogPaymentResponse(BaseModel):
    payment_status: str

File: frontend/src/api/payments.ts
import { api } from './client';

// payment_date/week are server-determined (log_payment()) — never client-supplied.
// note is optional free text Finance can attach (vendor detail modal's Pay flow).
export const postPayment = (vendorId: number, amount: number, note?: string) =>
  api.post<{ payment_status: string }>('/payments', { vendor_id: vendorId, amount, note: note || null });
